### ISSUE TITLE
Ads not being collected(caused by some rules in EasyList)

### ISSUE THREAD

**Issue by cqx931** [dhowe/AdNauseam#523]
Case: http://www.dailymail.co.uk/ushome/index.html

Ads can be correctly hidden and collected with just the adnauseam.txt file (with the right selector) enabled, but after enabling 'EasyList', ads can no longer be correctly collected.

**Potential Causes:**
1. A majority of blocks in AdNauseam come from 'EasyList', so it could be caused by some blocking rules in EasyList.To see logging messages from blocked requests, please comment out line 1042 in core.js (also set netLogging to 1). Note that a fix has been made to websocket blocking, so make sure you are using commit fc13e8f47b2f or newer.

2. Some cosmetic filters in EasyList may be hiding essential elements that break the ad-content generation process.

**Possible Directions:**
1.
`<div id="leader_lower_middle" class="adHolder" style="text-align: center;" data-adverts-ad="{data-here}"`
Data for the ad appears to be pulled from the div's 'data-adverts-ad' attribute; so some script may be running (and possibly being blocked by EasyList) that populates the div content from that data

2. The following rules in adnauseam.txt may be helpful:
```
dailymail.co.uk#@#.adHolder
dailymail.co.uk#@#.adsbygoogle
dailymail.co.uk#@#.billboard_wrapper
##iframe[id^=‘google_ads_iframe']
```

* Note: the performance of both hiding and collecting on this page is very unstable... sometimes I get no ads collected but everything hidden, sometimes even all ads showing (perhaps it is caused by different code in dynamically generated iframes? ) Anyway... now I can see the main banner being collected relatively constantly with the rules above. @leoneckert, please check whether it is true for you as well.

<img width="1255" alt="screen shot 2016-10-13 at 10 28 24 pm" src="https://cloud.githubusercontent.com/assets/3056154/19352986/7e682c0c-9194-11e6-8236-11e9e111f68e.png">

**Comment by dhowe** [comment]
> 2. The following rules in adnauseam.txt could be helpful:
> ```
> dailymail.co.uk#@#.adHolder
> dailymail.co.uk#@#.adsbygoogle
> dailymail.co.uk#@#.billboard_wrapper
...
```

The syntax of those rules don't look right to me... shouldn't it be:

` dailymail.co.uk##.adHolder`

**Comment by leoneckert** [comment]
isn't `dailymail.co.uk#@#.adHolder` an exception rule to NOT block the .adHolder elements, 
while `dailymail.co.uk##.adHolder` is an actual filter rule?

I took that over from what I learned from @cqx931 (she wrote si.com exceptions in adnauseam.txt using the same syntax)

**Comment by leoneckert** [comment]
@cqx931 I still don't seem to be able to collect the add. It appears with the three exception rules, but then `##iframe[id^=‘google_ads_iframe']` does not manage to hide/collect it.

**Comment by dhowe** [comment]
ok, though it seems to me we should first try to reliably collect the ad with hiding and blocking off (when everything is fully rendered).

**Comment by cqx931** [comment]
I just tried to select the ad with hiding and blocking off.
It seems like that the image is loaded after I get all the logs from adNauseam. (when I refresh the page, I instantly get some logs from the console, but the image shows up 1-2 seconds later)Therefore, when the filters are applied, there is indeed no image in the div yet.

Though when I disable easylist and other filters, the parser seems to waited longer to parese the filter?

**Comment by dhowe** [comment]
Interesting. I wonder if our MutationEvents (onDomChanged listeners) are firing correctly on this page. See domChangedHandler on line 1174

**Comment by cqx931** [comment]
When I have easylist enabled, the iframe that is added by addIframe() is still empty.
And when I disable easylist, we get the fully rendered iframe here.

**Comment by dhowe** [comment]
On hold -- revisit after release 2.4
